frontend/src/main/components/RecommendationRequest/RecommendationRequestForm.js

```diff
@@ -0,0 +1,160 @@
+    formState: { errors },
+    handleSubmit,
+  } = useForm({ defaultValues: initialContents || {} });
+  // Stryker restore all
+
+  const navigate = useNavigate();
+
+  // For explanation, see: https://stackoverflow.com/questions/3143070/javascript-regex-iso-datetime
+  // Note that even this complex regex may still need some tweaks
+
+  // Stryker disable Regex
+
+  // Stryker disable next-line all
+
+  return (
+    <Form onSubmit={handleSubmit(submitAction)}>
+      <Row>
+        {initialContents && (
+          <Col>
+            <Form.Group className="mb-3">
+              <Form.Label htmlFor="id">Id</Form.Label>
+              <Form.Control
+                data-testid="RecommendationRequestForm-id"
+                id="id"
+                type="text"
+                {...register("id")}
+                value={initialContents.id}
+                disabled
+              />
+            </Form.Group>
+          </Col>
+        )}
+      </Row>
+
+      <Row>
+        <Col>
+          <Form.Group className="mb-3">
+            <Form.Label htmlFor="professorName">Professor Name</Form.Label>
+            <Form.Control
+              data-testid="RecommendationRequestForm-professorName"
+              id="professorName"
+              type="text"
+              isInvalid={Boolean(errors.professorName)}
+              {...register("professorName", {
+                required: "professorName is required.",
+              })}
+            />
+            <Form.Control.Feedback type="invalid">
+              {errors.professorName?.message}
+            </Form.Control.Feedback>
+          </Form.Group>
+        </Col>
+        <Col>
+          <Form.Group className="mb-3">
+            <Form.Label htmlFor="professorEmail">Professor Email</Form.Label>
+            <Form.Control
+              data-testid="RecommendationRequestForm-professorEmail"
+              id="professorEmail"
+              type="text"
+              isInvalid={Boolean(errors.professorEmail)}
+              {...register("professorEmail", {
+                required: "professorEmail is required.",
+              })}
+            />
+            <Form.Control.Feedback type="invalid">
+              {errors.professorEmail?.message}
+            </Form.Control.Feedback>
+          </Form.Group>
+        </Col>
+        <Col>
+          <Form.Group className="mb-3">
+            <Form.Label htmlFor="requesterName">Requester Name</Form.Label>
+            <Form.Control
+              data-testid="RecommendationRequestForm-requesterName"
+              id="requesterName"
+              type="text"
+              isInvalid={Boolean(errors.requesterName)}
+              {...register("requesterName", {
+                required: "requesterName is required.",
+              })}
+            />
+            <Form.Control.Feedback type="invalid">
+              {errors.requesterName?.message}
+            </Form.Control.Feedback>
+          </Form.Group>
+        </Col>
+      </Row>
```

Comment:
The POST endpoint doesn't ask for this information because the RequesterId comes from the currentUser.

Again: when creating a form for a new record, you need to coordinate with the POST endpoint in the backend, even if you aren't hooking up the form to that POST endpoint yet.

Reply:
ah, just realized that I was going off of a previous version of the controller. just to be clear, the parameters of the controller are professorEmail, recommendationType, and details, which are the only three fields I should include on the form.